config.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"flag"
+
+	"github.com/justsocialapps/holmes/analytics"
+)
+
+var (
+	protocol     string
+	host         string
+	proxyPort    string
+	proxyPath    string
+	listenHost   string
+	listenPort   string
+	kafkaHost    string
+	logfileName  string
+	printVersion bool
+	anomyzeIP    bool
+)
```

Comment:
Ist das echt üblich in Go einfach so über mehrere Dateien verteilt Variablen zu definieren und zu benutzen? Ich fände es irgendwie schöner, dass in eine struct zu packen, die dann in der main()-Funktion explizit erzeugt wird. So ist mir das irgendwie zu magisch.

Ist aber vielleicht auch einfach nur, weil ich in Go nicht so drin bin.

(Aber grundsätzlich finde ich das sehr gut das auszulagern!)

Reply:
Das ist in Go tatsächlich gängig und wird häufig sogar in ein extra Package ausgelagert.
Meistens wird hierbei aber auch auf Config-Tools wie Viper zurück gegriffen. Dies fand ich aber etwas zu viel des Guten.